.pipelines/build-linux.yml

```diff
@@ -0,0 +1,32 @@
+pr:
+- master
+
+pool:
+  vmImage: 'ubuntu-16.04'
+
+container:
+  image: vowpalwabbit/ubuntu1604-build:0.1.0
+  endpoint: DockerHub
+
+steps:
+- bash: ./.scripts/linux/clang-format-check.sh $(Build.SourcesDirectory)
```

Comment:
Is there a reason why you are relying on passing Build.SourcesDirectory to each of the scripts? It seems like it would make it more difficult to use those scripts locally if they relied on relative structure.

Reply:
The scripts accept the repository directory as their only argument so it can know the base path and find the things it needs

Reply:
Would it be worthwhile to do something similar to how init.cmd detects its location, then sets up the various environment variables that the rest of the build scripts depend on?

From https://stackoverflow.com/questions/59895/get-the-source-directory-of-a-bash-script-from-within-the-script-itself:

`DIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" >/dev/null 2>&1 && pwd )"`

Reply:
This is a cool solution, I like it!